File: 0024 两两交换链表中的节点/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if not head or not head.next:
            return head
        root=head.next
        m,n=head ,head.next
        lastn=ListNode(0)
        while m.next and n.next :
            m.next=n.next
            n.next=m
            lastn.next=n
            lastn=m
            m=m.next
            n=m.next
        if(n):
            m.next=n.next
            n.next=m
            lastn.next=n
        return root

File: 0024 两两交换链表中的节点/main.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode hhead(0,head);
        ListNode *node1=&hhead;
        ListNode *node2=head;
        ListNode *node3;
        while( node1->next !=nullptr && node1->next->next !=nullptr ){
            node2=node1->next;
            node3=node2->next;

            node2->next=node3->next;
            node3->next=node2;
            node1->next=node3;

            node1=node1->next->next;
        }
        return hhead.next;
    }
};